Assets/Mirror/Components/NetworkTransformSnapshotInterpolation.cs

```diff
@@ -0,0 +1,351 @@
+                }
+                else
+                {
+                    ClientInterpolation();
+                }
+            }
+        }
+
+        #region Server Sync Update
+        void ServerSyncUpdate()
+        {
+            if (ServerNeedsToSendUpdate())
+            {
+                SendMessageToClient();
+                ClearNeedsUpdate(clientSyncInterval);
+            }
+        }
+
+        /// <summary>
+        /// Checks if object needs syncing to clients
+        /// <para>Called on server</para>
+        /// </summary>
+        /// <returns></returns>
+        bool ServerNeedsToSendUpdate()
+        {
+            if (IsControlledByServer)
+            {
+                return IsTimeToUpdate() && (HasMoved() || HasRotated());
+            }
+            else
+            {
+                // dont care about time here, if client authority has sent snapshot then always relay it to other clients
+                // todo do we need a check for attackers sending too many snapshots?
+                return _needsUpdate;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        void SendMessageToClient()
+        {
+            // if client has send update, then we want to use the state it gave us
+            // this is to make sure we are not sending host's interpolations position as the snapshot insteading sending the client auth snapshot
+            TransformState state = IsControlledByServer ? TransformState : _latestState;
+            // todo is this correct time?
+            RpcServerSync(state, NetworkTime.time);
+        }
+
+        [ClientRpc]
+        void RpcServerSync(TransformState state, double time)
+        {
+            // not host
+            // host will have already handled movement in servers code
+            if (isServer)
+                return;
+
+            AddSnapShotToBuffer(state, time);
+        }
+
+        /// <summary>
+        /// Applies values to target transform on client
+        /// <para>Adds to buffer for interpolation</para>
+        /// </summary>
+        /// <param name="state"></param>
+        void AddSnapShotToBuffer(TransformState state, double serverTime)
+        {
+            // dont apply on local owner
+            if (IsLocalClientInControl)
+                return;
+
+            snapshotBuffer.AddSnapShot(state, serverTime);
+        }
+        #endregion
+
+
+        #region Client Sync Update 
+        void ClientAuthorityUpdate()
+        {
+            if (IsTimeToUpdate() && (HasMoved() || HasRotated()))
+            {
+                SendMessageToServer();
+                ClearNeedsUpdate(clientSyncInterval);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        void SendMessageToServer()
+        {
+            // todo, is this the correct time?
+            CmdClientAuthoritySync(TransformState, NetworkTime.time);
+        }
+
+        [Command]
```

Comment:
```suggestion
        [Command(channel = Channels.DefaultUnreliable)]
```
Default unreliable for transports that support it.

Reply:
unreliable sends is in the "Future improvements" part

Reply:
ok but of all that's needed is that channel parameter, why not?

Reply:
I've only tested it with telepathy, I dont want to add untested behaviour. If we add `channel = Channels.DefaultUnreliable` now users will assume it is tested and behaves well.

2 other things to explain my reasoning:
- This is an MVP, it is not perfect of feature complete. The goal is to make something better than the old NT for most people (but not all people, eg some need teleport function or scale from old NT)
- I dont want to just add stuff for the sake of adding it even if it is just 1 line, that will lead to bloat.

for the case of "unreliable sends" it is in the "Future improvements", or in other words, Not in MVP, but planned for future.